lib/generators/apple_auth/apple_sign_in_controller/templates/apple_sign_in_controller.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class <%= @scope_prefix %>AppleSignInController < DeviseTokenAuth::SessionsController
+  protect_from_forgery with: :null_session
+  skip_before_action :verify_authenticity_token
+  before_action :skip_session_storage
+  before_action :check_json_request
+
+  def apple_sign_in
+    apple_params = apple_validate
+    @resource = sign_in_with_apple(apple_params)
+    custom_sign_in
+  rescue AppleAuth::Conditions::JWTValidationError => e
+    render_error(:bad_request, e.message)
+  rescue OAuth2::Error => e
+    render_error(:bad_request, e.message)
+  rescue JWT::ExpiredSignature => e
+    render_error(:bad_request, e.message)
+  end
+
+  private
+
+  def apple_validate
+    data = AppleAuth::UserIdentity.new(
+      apple_sign_in_params[:user_identity],
+      apple_sign_in_params[:jwt]
+    ).validate!
+    AppleAuth::Token.new(apple_sign_in_params[:code]).authenticate
```

Comment:
Probably in another PR but to keep consistency, could we call this method `authenticate!`? What do you think @TimoPeraza ? Because it could raise exceptions right?

Reply:
Yea, that sounds good